Mathlib/RingTheory/Noetherian.lean

```diff
@@ -619,3 +619,21 @@ theorem IsNoetherianRing.isNilpotent_nilradical (R : Type*) [CommRing R] [IsNoet
   obtain ⟨n, hn⟩ := Ideal.exists_radical_pow_le_of_fg (⊥ : Ideal R) (IsNoetherian.noetherian _)
   exact ⟨n, eq_bot_iff.mpr hn⟩
 #align is_noetherian_ring.is_nilpotent_nilradical IsNoetherianRing.isNilpotent_nilradical
+
+theorem isNilpotent_iff_of_Finite (R M : Type*) [CommRing R] [AddCommGroup M] [Module R M]
+    [Module.Finite R M] (f : M →ₗ[R] M) :
+    IsNilpotent f ↔ ∀ m : M, ∃ n : ℕ, (f^n) m = 0 := by
```

Comment:
would Module.End.isNilpotent_iff_of_Finite be an appropriate name? It seems that it should be to me, since this statement is about the nilpotency of an endomorphism as you point out.